controls/gdal_works.rb

```diff
@@ -19,13 +19,180 @@
+      its('stdout') { should_not be_empty }
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
+  gdaltransform_full_path = File.join(plan_installation_directory.stdout.strip, "bin/gdaltransform")
+  describe command("#{gdaltransform_full_path} --version") do
+      its('exit_status') { should eq 0 }
+      its('stdout') { should_not be_empty }
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
+  gdalwarp_full_path = File.join(plan_installation_directory.stdout.strip, "bin/gdalwarp")
+  describe command("#{gdalwarp_full_path} --version") do
+      its('exit_status') { should eq 0 }
+      its('stdout') { should_not be_empty }
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
+  gnmanalyse_full_path = File.join(plan_installation_directory.stdout.strip, "bin/gnmanalyse")
+  describe command("#{gnmanalyse_full_path} --version") do
+      its('exit_status') { should eq 0 }
+      its('stdout') { should_not be_empty }
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
+  gnmmanage_full_path = File.join(plan_installation_directory.stdout.strip, "bin/gnmmanage")
+  describe command("#{gnmmanage_full_path} --version") do
+      its('exit_status') { should eq 0 }
+      its('stdout') { should_not be_empty }
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
+  nearblack_full_path = File.join(plan_installation_directory.stdout.strip, "bin/nearblack")
+  describe command("#{nearblack_full_path} --version") do
+      its('exit_status') { should eq 0 }
+      its('stdout') { should_not be_empty }
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
+  ogr2ogr_full_path = File.join(plan_installation_directory.stdout.strip, "bin/ogr2ogr")
+  describe command("#{ogr2ogr_full_path} --version") do
+      its('exit_status') { should eq 0 }
+      its('stdout') { should_not be_empty }
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
+  ogrinfo_full_path = File.join(plan_installation_directory.stdout.strip, "bin/ogrinfo")
+  describe command("#{ogrinfo_full_path} --version") do
+      its('exit_status') { should eq 0 }
+      its('stdout') { should_not be_empty }
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
+  ogrlineref_full_path = File.join(plan_installation_directory.stdout.strip, "bin/ogrlineref")
+  describe command("#{ogrlineref_full_path} --version") do
+      its('exit_status') { should eq 0 }
+      its('stdout') { should_not be_empty }
+      its('stdout') { should match /GDAL #{plan_pkg_version}/ }
+      its('stderr') { should be_empty }
+  end
```

Comment:
Can you consolidate and loop over these tests? See example [here](https://github.com/chef-base-plans/python2/blob/07207f3ce5f3fe713ab59d9795358a43a65911c0/controls/python_works.rb#L69-L75)

Reply:
All 'works' tests combined into single array loop apart from a few exceptions that need slight variations on the checks